src/lib/snark_work_lib/partitioned.ml

```diff
@@ -555,53 +555,37 @@ module Result = struct
 
     module V1 = struct
       type t =
-        { data :
-            ( Transaction_witness.Stable.V2.t
-            , Transaction_snark.Zkapp_command_segment.Witness.Stable.V1.t
-            , Ledger_proof.Stable.V2.t
-            , Proof_with_metric.Stable.V1.t )
-            Spec.Poly.Stable.V1.t
-        ; prover : Signature_lib.Public_key.Compressed.Stable.V1.t
-        }
+        ( Transaction_witness.Stable.V2.t
+        , Transaction_snark.Zkapp_command_segment.Witness.Stable.V1.t
+        , Ledger_proof.Stable.V2.t
+        , Proof_with_metric.Stable.V1.t )
+        Poly.Stable.V1.t
 
       let to_latest = Fn.id
     end
   end]
 
   type t =
-    { data :
-        ( Transaction_witness.t
-        , Transaction_snark.Zkapp_command_segment.Witness.t
-        , Ledger_proof.Cached.t
-        , Proof_with_metric.t )
-        Spec.Poly.t
-    ; prover : Signature_lib.Public_key.Compressed.t
-    }
-
-  let read_all_proofs_from_disk ({ data; prover } : t) : Stable.Latest.t =
-    let data =
-      Spec.Poly.map ~f_witness:Transaction_witness.read_all_proofs_from_disk
-        ~f_zkapp_command_segment_witness:
-          Transaction_witness.Zkapp_command_segment_witness
-          .read_all_proofs_from_disk
-        ~f_proof:Ledger_proof.Cached.read_proof_from_disk
-        ~f_metric:Proof_with_metric.read_all_proofs_from_disk data
-    in
-
-    { data; prover }
+    ( Transaction_witness.t
+    , Transaction_snark.Zkapp_command_segment.Witness.t
+    , Ledger_proof.Cached.t
+    , Proof_with_metric.t )
+    Poly.t
 
-  let write_all_proofs_to_disk ~proof_cache_db
-      ({ data; prover } : Stable.Latest.t) : t =
-    let data =
-      Spec.Poly.map
-        ~f_witness:
-          (Transaction_witness.write_all_proofs_to_disk ~proof_cache_db)
-        ~f_zkapp_command_segment_witness:
-          (Transaction_witness.Zkapp_command_segment_witness
-           .write_all_proofs_to_disk ~proof_cache_db )
-        ~f_proof:(Ledger_proof.Cached.write_proof_to_disk ~proof_cache_db)
-        ~f_metric:(Proof_with_metric.write_all_proofs_to_disk ~proof_cache_db)
-        data
-    in
-    { data; prover }
+  let read_all_proofs_from_disk =
```

Comment:
This function is still wrong. Fixed https://github.com/MinaProtocol/mina/pull/17134/commits/8f237b65cf271ef871d7cf404cf4d3626879f6db

Didn't back port it because it's only 1 commit.